generate_analysis_graphs.py

```diff
@@ -112,3 +113,33 @@
         )
         chart.save(f"{output_dir}/season_distribution_{plan.analysis_file_key}.html")
         chart.save(f"{output_dir}/season_distribution_{plan.analysis_file_key}.png", scale_factor=IMG_SCALE_FACTOR)
+
+    # Compute the number of UIDs with manually labelled relevant messages per show
+    log.info("Graphing the No. of UIDs with relevant messages per show...")
+    relevant_uids_per_show = OrderedDict()
+
+    for plan in PipelineConfiguration.RQA_CODING_PLANS:
+        relevant_uids_per_show[plan.raw_field] = 0
+    for msg in messages:
+        for plan in PipelineConfiguration.RQA_CODING_PLANS:
+            if msg["consent_withdrawn"] == Codes.TRUE:
+                continue
+            if plan.binary_code_scheme is not None:
+                binary_coda_code = plan.binary_code_scheme.get_code_with_id(msg[plan.binary_coded_field]["CodeID"])
```

Comment:
This needs factoring into a function in core at some point so that we can validate and test the algorithm, and use it consistently with other parts as well. @IsaackMwenda could you chat with @as2388 about where is best to put it

Reply:
Noted thank you.